src/services/Xml.php

```diff
@@ -496,7 +496,10 @@ protected function addChildWithCDATA(&$xml, $name, $value)
     {
         $new_child = $xml->addChild($name);
         if ($new_child !== null) {
-            $node = dom_import_simplexml($new_child);
+			$node = dom_import_simplexml($new_child);
```

Comment:
You should be able to revert this, and instead on line [201-203](https://github.com/FosterCommerce/shipstation-connect/pull/6/files#diff-baff5883d2219e1e7dfe686e4724d017R202), that if statement and its body can be removed. I think a lot of the issue you experienced is to do with the snapshot JSON plus the JSON you've added to order items.